db/redis.go

```diff
@@ -248,30 +266,144 @@ func (r *RedisDriver) InsertKEVulns(records []models.KEVuln) (err error) {
 	return nil
 }
 
-// GetKEVulnByCveID :
-func (r *RedisDriver) GetKEVulnByCveID(cveID string) ([]models.KEVuln, error) {
+// InsertVulnCheck :
+func (r *RedisDriver) InsertVulnCheck(records []models.VulnCheck) (err error) {
+	ctx := context.Background()
+	batchSize := viper.GetInt("batch-size")
+	if batchSize < 1 {
+		return fmt.Errorf("Failed to set batch-size. err: batch-size option is not set properly")
+	}
+
+	// newDeps, oldDeps: {"CVEID": {"VulnCheck:HashSum(CVEJSON)": {}}}
+	newDeps := map[string]map[string]struct{}{}
+	oldDepsStr := "{}"
+	t, err := r.conn.Type(ctx, depKey).Result()
+	if err != nil {
+		return xerrors.Errorf("Failed to Type key: %s. err: %w", depKey, err)
+	}
+	switch t {
+	case "string":
+		if _, err := r.conn.Del(ctx, depKey).Result(); err != nil {
```

Comment:
In this case, should the value be inserted to KEV#DEP hash with CISA key after deletion?

Reply:
fixed

https://github.com/vulsio/go-kev/compare/72ea3f9224d5d49c9cd8a6a5ddbcf2bff9343f31..f0e9def18b52f6d4133eb9325dadb9d6ecd93e8b